contracts/utils/deploy.js

```
@@ -1651,6 +1651,80 @@ function encodeSaltForCreateX(deployer, crossChainProtectionFlag, salt) {
   return encodedSalt;
 }
 
+async function createPoolBoosterSonic({
+  cOSonic,
+  factoryContract,
+  pools,
+  salt,
+  split = null,
+  type = "Single", // "Single" or "Double"
+  signatureSingle = "createPoolBoosterSwapxSingle(address,address,uint256)",
+  signatureDouble = "createPoolBoosterSwapxDouble(address,address,address,uint256,uint256)",
+}) {
+  const poolBoosterCreationArgs = {};
+  const poolBoosterComputedAddresses = {};
+
+  const getAddress = (path) =>
+    path.split(".").reduce((obj, key) => obj?.[key], addresses.sonic);
+
+  await Promise.all(
+    pools.map(async (pool) => {
+      const current = getAddress(pool);
+      if (!current?.extBribeOS || !current?.pool) return;
+
+      const args =
+        type === "Single"
+          ? [current.extBribeOS, current.pool, salt]
+          : [
+              current.extBribeOS,
+              current.extBribeOther,
+              current.pool,
+              split,
+              salt,
+            ];
+
+      if (args.some((arg) => arg === undefined)) return;
+
+      poolBoosterCreationArgs[pool] = args;
+
+      poolBoosterComputedAddresses[pool] =
+        await factoryContract.computePoolBoosterAddress(
+          ...poolBoosterCreationArgs[pool]
+        );
+
+      console.log(
+        `Pool Booster Swapx ${type} ${pool} computed address: ${poolBoosterComputedAddresses[pool]}`
+      );
+    })
+  );
+
+  const actions = pools.flatMap((pool) => {
+    const current = getAddress(pool);
+    if (!current?.pool || !poolBoosterComputedAddresses[pool]) return [];
```

should we throw an exception here also?